src/Nethermind/Nethermind.State/StateProvider.cs

```diff
@@ -59,6 +59,8 @@ public void RecalculateStateRoot()
         {
             _tree.UpdateRootHash();
             _needsStateRootUpdate = false;
+
+            StateRootCommitted?.Invoke(this, new StateRootCommittedEventArgs(_tree.RootHash));
```

Comment:
Not sure about naming as it can also be invoked in non-comited scenarios.

Reply:
yeah, will change it.